tests/bls_tests.rs

```diff
@@ -131,34 +100,26 @@ impl AggVerifyTestIO {
         let signature: Signature =
             match Signature::from_bytes(&decode_hex_with_prefix(&self.input.signature)) {
                 Ok(sign) => sign,
-                Err(_) => {
-                    // handling for the zero signature case which raises a blst bad encoding error
-                    if self.output == false {
-                        return true;
-                    } else {
-                        return false;
-                    }
-                }
+                // handling for the zero signature case which raises a blst bad encoding error
+                Err(_) => return assert_eq!(self.output, false),
```

Comment:
i think you want this:

```suggestion
                Err(_) => return self.output == false,
```

Reply:
and this starts to get a little opaque but `return self.output == false` should be the same as `return !self.output` right?

Reply:
Yes, but `==` returns `bool` but we want `()` which is returned by `assert_eq!`.

Reply:
Ah, I'll change it to that then.